### PR TITLE
Update installer.sh to support the ANDROID_HOME env automation

### PR DESCRIPTION
**Pull request by hydraxman**
## Description
Update make it easier for user to install agent on MacOS.

## Testing

Tested on my MacOS. This is hard to test. So we might need to figure out a way for testing and ensuring the quality of the script, this `installer.sh` is a part of the user experience as well and we need to make sure its accountability.

### CODE REVIEW

File: agent/agent_installer/MacOS/iOS/installer.sh [hunk 1]
```diff
@@ -15,6 +15,20 @@ echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> "$userHome/.zprofile"
 eval "$(/opt/homebrew/bin/brew shellenv)"
 brew install carthage
 brew install node
+brew install awk
+brew install android-platform-tools
+
+# trace adb path and set ANDROID_HOME as env var
+adb_path=`which adb`
+adb_path=`ls -l $adb_path`
```
Comment (hydraxman) [inline agent/agent_installer/MacOS/iOS/installer.sh:23]:
```suggestion
# example value of this adb_path here: 
# lrwxr-xr-x  1 bsp  admin  71 Mar 10 18:47 /opt/homebrew/bin/adb -> /opt/homebrew/Caskroom/android-platform-tools/34.0.0/platform-tools/adb

adb_path=`ls -l $adb_path`
```

File: agent/agent_installer/MacOS/iOS/installer.sh [hunk 2]
```diff
@@ -15,6 +15,22 @@ echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> "$userHome/.zprofile"
...
+adb_path=`which adb`
+# example value of this adb_path here: 
+# lrwxr-xr-x  1 bsp  admin  71 Mar 10 18:47 /opt/homebrew/bin/adb -> /opt/homebrew/Caskroom/android-platform-tools/34.0.0/platform-tools/adb
+adb_path=`ls -l $adb_path`
+echo $adb_path > temp.txt
+adb_path=`awk -F '-> ' '{print $2}' temp.txt`
+echo $adb_path > temp.txt
+export ANDROID_HOME=`awk -F '/platform-tools/' '{print $1}' temp.txt`
+echo $ANDROID_HOME
+echo "\nexport ANDROID_HOME=$ANDROID_HOME" >> export_android_home.sh
```
Comment (dinghong2233) [inline agent/agent_installer/MacOS/iOS/installer.sh:31]: nit: Should we remove the temp.txt and export_android_home.sh after the env set up.

Comment (hydraxman) [reply agent/agent_installer/MacOS/iOS/installer.sh]: I will remove temp.txt and keep export_android_home.sh for debugging.

**Review by dinghong2233 — DISMISSED** [review]

File: agent/agent_installer/MacOS/iOS/installer.sh [hunk 3]
```diff
@@ -63,22 +93,24 @@ end tell'
...
 
 ### copy files 
-echo "Copying Files......"
-echo "Work Folder: $serverPath"
+echo "Copying Files to Work Folder: $serverPath"
 mkdir -p "$serverPath"
 sudo mkdir -p "$supervisorPath"
 sudo cp "$iniFile" "$supervisorPath/HydraLabAgent.ini"
 jarPath=$(find "$scriptPath" -type f -name "*agent*")
+
```
Comment (hydraxman) [inline agent/agent_installer/MacOS/iOS/installer.sh:109]:
```suggestion
```

**Review by zhou9584 — APPROVED** [review]

**Review by dinghong2233 — APPROVED** [review]

**Pull request merged**